# 18.6 函數

如何使用內建函數`recover`終止`panic`過程（參考[章節13.3](13.3.md)）：
    
```go
func protect(g func()) {
    defer func() {
        log.Println("done")
        // Println executes normally even if there is a panic
        if x := recover(); x != nil {
            log.Printf("run time panic: %v", x)
        }
    }()
    log.Println("start")
    g()
}
```

## 鏈接

- [目錄](directory.md)
- 上一節：[接口](18.5.md)
- 下一節：[文件](18.7.md)